var/spack/repos/builtin/packages/3dtk/package.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class _3dtk(CMakePackage):
+    """The 3D Toolkit provides algorithms and methods to process 3D point
+    clouds.
+
+    It includes automatic high-accurate registration (6D simultaneous
+    localization and mapping, 6D SLAM) and other tools, e.g., a fast 3D viewer,
+    plane extraction software, etc. Several file formats for the point clouds
+    are natively supported, new formats can be implemented easily."""
+
+    homepage = "http://slam6d.sourceforge.net/"
+    # Repo seems to be in the process of switching to git:
+    # https://github.com/3DTK/3DTK
+
+    version('trunk', svn='https://svn.code.sf.net/p/slam6d/code/trunk')
+    version('1.2', svn='https://svn.code.sf.net/p/slam6d/code/branches/3dtk-release-1.2')
+
+    variant('cgal', default=False, description='Compile with CGAL support')
+    variant('opengl', default=True, description='Compile with OpenGL support')
+    variant('opencv', default=True, description='Compile with OpenCV support')
+    variant('compact_octree', default=False, description='Whether to use the compact octree display')
+    variant('cuda', default=False, description='Whether to build CUDA accelerated collision detection tools')
+    variant('openmp', default=False, description='Whether to use parallel processing capabilities of OPENMP')
+
+    conflicts('~opencv', when='platform=darwin')
+    conflicts('+compact_octree', when='~opengl')
+
+    generator = 'Ninja'
+
+    depends_on('cmake@3.5:', type='build')
+    depends_on('ninja', type='build')
+    depends_on('boost+serialization+graph+regex+filesystem+system+thread+date_time+program_options')
+    depends_on('suite-sparse')
+    depends_on('zlib')
+    depends_on('libpng')
+    depends_on('cgal', when='+cgal')
+    depends_on('gl', when='+opengl')
+    depends_on('glew', when='+opengl')
+    depends_on('freeglut', when='+opengl')
+    depends_on('opencv+calib3d+core+features2d+highgui+imgproc+ml', when='+opencv')
```

Comment:
```suggestion
    depends_on('opencv+calib3d+core+features2d+highgui+imgcodecs+imgproc+ml+videoio', when='+opencv')
```
I needed to add `imgcodecs` and `videoio`, otherwise functions were missing.